CRM/Case/BAO/Case.php

```diff
@@ -3168,4 +3167,58 @@ public function addSelectWhereClause() {
     return $clauses;
   }
 
+  /**
+   * CRM-20308
+   * Method to get the contact id to use as from contact for email copy
+   * 1. Activity Added by Contact's email address
+   * 2. System Default From Address
+   * 3. Default Organization Contact email address
+   * 4. Logged in user
+   *
+   * @param int $activityId
+   * @return mixed $emailFromContactId
+   * @see https://issues.civicrm.org/jira/browse/CRM-20308
+   */
+  private static function getReceiptFrom($activityId) {
+    $emailFromContactId = NULL;
+    if (!empty($activityId)) {
+      try {
+        $sourceRecordTypeId = civicrm_api3('OptionValue', 'getvalue', array(
+          'option_group_id' => 'activity_contacts',
+          'name' => 'Activity Source',
+          'return' => 'value'));
+        $emailFromContactId = civicrm_api3('ActivityContact', 'getvalue', array(
+          'activity_id' => $activityId,
+          'record_type_id' => $sourceRecordTypeId,
```

Comment:
The api will resolve this (with better caching) if you do 

'record_type_id' => 'Activity Source'